hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AsyncCleanerService.java

```diff
@@ -73,6 +73,8 @@ public static void waitForCompletion(AsyncCleanerService asyncCleanerService) {
         asyncCleanerService.waitForShutdown();
       } catch (Exception e) {
         throw new HoodieException("Error waiting for async cleaning to finish", e);
+      } finally {
+        asyncCleanerService.shutdown(false);
       }
```

Comment:
Guess this is the leak

Reply:
Actually no, it is the monitor thread pool leaks !